### ISSUE TITLE
Race condition when replacing Google Plus social widget

### ISSUE THREAD

**Issue by gunesacar** [EFForg/privacybadgerchrome#404]
It seems Privacy Badger sometimes fails to replace Google Plus (+1) widget due to a race condition. This was the case before the pull request #395 and will resurface after the fix for #402 gets merged.

To reproduce, apply the patch at #402 and visit a page that includes a +1 button (e.g. [1](http://www.hln.be/hln/nl/1326/Eurosong/article/detail/2333906/2015/05/22/Loic-Nottet-van-outsider-naar-regelrechte-kanshebber-het-loopt-storm-op-iTunes.dhtml), [2](http://www.gamasutra.com/blogs/LarsDoucet/20131203/206159/On_Procedural_Death_Labyrinths.php), [3](http://www.snapdeal.com/))
If you reload the pages multiple times you should be able to observe cases where +1 button is not replaced.

I guess this is because PB's selectors try to match [+1's placeholder elements](https://developers.google.com/+/web/+1button/#example-basic) (`g:plusone` and `div.g-plusone`) that are removed when the button is rendered by the associated script. If `replaceInitialTrackerButtonsHelper` checks the page after the button is rendered, PB cannot find anything to replace.

The failure is more common on beefy pages linked above, while +1 button is always gets replaced on ~blank test pages I setup myself ([1](https://securehomes.esat.kuleuven.be/~gacar/dev/pb/test/gplus_basic.html), [2](https://securehomes.esat.kuleuven.be/~gacar/dev/pb/test/gplus_explicit_render.html), [3](https://securehomes.esat.kuleuven.be/~gacar/dev/pb/test/gplus_explicit_load.html)).

**Comment by SwartzCr** [comment]
fixed in #403

**Comment by gunesacar** [comment]
This issue is not fixed in #403, rather reappeared after #403, as expected (see [my first comment](https://github.com/EFForg/privacybadgerchrome/issues/404#issue-82946762)).

The issue can still be observed on the links shared above (e.g. [1](http://www.hln.be/hln/nl/1326/Eurosong/article/detail/2333906/2015/05/22/Loic-Nottet-van-outsider-naar-regelrechte-kanshebber-het-loopt-storm-op-iTunes.dhtml), [2](http://www.gamasutra.com/blogs/LarsDoucet/20131203/206159/On_Procedural_Death_Labyrinths.php), [3](http://www.snapdeal.com/)). You may need to reload many times until you get a case where +1 button is not replaced.

**Comment by ghostwords** [comment]
I'm going to mark this as a duplicate of #788.

**Comment by ghostwords** [comment]
I don't think this was fixed as part of the [May 2016 release](https://github.com/EFForg/privacybadger/compare/chrome-2016.4.13...chrome-2016.5.16) ...
